smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpRpcProtocolGenerator.java

```diff
@@ -420,22 +433,57 @@ private void generateErrorDeserializer(GenerationContext context, StructureShape
+            StructureShape outputShape = context.getModel().expectShape(shapeId).asStructureShape().get();
+
+            Collection<MemberShape> memberValues = outputShape.getAllMembers().values();
+            Optional<MemberShape> first = memberValues.stream().findFirst();
+            Set<UnionShape> unionShapesWithTrait =
+                    context.getModel().getUnionShapesWithTrait(StreamingTrait.class);
+            boolean hasStreamingUnion = first.isPresent()
+                    && memberValues.size() == 1
+                    && unionShapesWithTrait
+                    .stream()
+                    .map(Shape::getId)
+                    .collect(Collectors.toSet())
+                    .contains(first.get().getTarget());
+
+            return hasStreamingUnion;
+        }
+        return false;
+    }
+
     private void readResponseBody(GenerationContext context, OperationShape operation) {
         TypeScriptWriter writer = context.getWriter();
         OptionalUtils.ifPresentOrElse(
                 operation.getOutput(),
                 outputId -> {
                     // We only need to load the body and prepare a contents object if there is a response.
-                    writer.write("const data: any = await parseBody(output.body, context)");
-                    writer.write("let contents: any = {};");
-
                     // If there's an output present, we know it's a structure.
                     StructureShape outputShape = context.getModel().expectShape(outputId).asStructureShape().get();
 
-                    // Track output shapes so their deserializers may be generated.
-                    deserializingDocumentShapes.add(outputShape);
-
-                    deserializeOutputDocument(context, operation, outputShape);
+                    Collection<MemberShape> memberValues = outputShape.getAllMembers().values();
+                    Optional<MemberShape> first = memberValues.stream().findFirst();
+                    boolean useEventStreamDeserializer = isStreamingUnion(context, operation);
+
+                    if (useEventStreamDeserializer) {
+                        writer.write("const contents: any = {};");
+                        HttpProtocolGeneratorUtils.generateRpcEventStreamDeserializer(
+                                context,
+                                outputShape,
+                                first.get().getMemberName(),
+                                "contents." + first.get().getMemberName()
+                        );
+                        deserializingDocumentShapes.add(outputShape);
+                    } else {
+                        writer.write("const data: any = await parseBody(output.body, context)");
+                        // Track output shapes so their deserializers may be generated.
+                        deserializingDocumentShapes.add(outputShape);
+                        writer.write("let contents: any = {};");
```

Comment:
move this 2 lines out of the if-else statements.

Reply:
I'll move the `.add` but the other one is let vs const